.sphinx/lxd-doc/README.md

```diff
@@ -0,0 +1,177 @@
+	// ---
+	//  shortdesc: Possible values are all, manual and group. See Automatic placement of instances for more information.
+	//  <ANY_KEY>: <ANY_VALUE>
+    //  condition: container
+	//  default: `all`
+	//  type: integer
+	//  liveupdate: `yes`
+    clusterConfigKeys := map[string]func(value string) error{
+		"scheduler.instance": validate.Optional(validate.IsOneOf("all", "group", "manual")),
+	}
+
+    for k, v := range config {
+		// lxddoc:generate(group=cluster, key=user.*)
+		//
+		// This is the real long desc.
+		//
+		// With two paragraphs.
+		//
+		// And a list:
+		//
+		// - Item
+		// - Item
+		// - Item
+		//
+		// example of a table:
+		//
+		// Key                                 | Type      | Scope     | Default                                          | Description
+		// :--                                 | :---      | :----     | :------                                          | :----------
+		// `acme.agree_tos`                    | bool      | global    | `false`                                          | Agree to ACME terms of service
+		// `acme.ca_url`                       | string    | global    | `https://acme-v02.api.letsencrypt.org/directory` | URL to the directory resource of the ACME service
+		// `acme.domain`                       | string    | global    | -                                                | Domain for which the certificate is issued
+		// `acme.email`                        | string    | global    | -                                                | Email address used for the account registration
+		//
+		//  ---
+		//	shortdesc: Free form user key/value storage (can be used in search).
+		//	condition: container
+		//	default: -
+		//	type: string
+		//	liveupdate: `yes`
+		if strings.HasPrefix(k, "user.") {
+			continue
+		}
+
+		validator, ok := clusterConfigKeys[k]
+		if !ok {
+			return fmt.Errorf("Invalid cluster configuration key %q", k)
+		}
+
+		err := validator(v)
+		if err != nil {
+			return fmt.Errorf("Invalid cluster configuration key %q value", k)
+		}
+	}
+
+	return nil
+```
+
+The go-swagger spec from source generator can only handles `swagger:meta` (global file/package level documentation), `swagger:route` (API endpoints), `swagger:params` (function parameters), `swagger:operation` (method documentation), `swagger:response` (API response content documentation), `swagger:model` (struct documentation) generation. In our use case, we would want a config variable spec generator that can bundle any key-value data pairs alongside metadata to build a sense of hierarchy and identity (we want to associate a unique key to each lxddoc comment group that will also be displayed in the generated documentation)
+
+In a swagger fashion, `lxddoc` can associate metadata key-value pairs (here for example, `group` and `key`) to data key-value pairs. As a result, it can generate a YAML tree out of the code documentation and also a Markdown document.
```

Comment:
```suggestion
In a swagger fashion, `lxd-doc` can associate metadata key-value pairs (here for example, `group` and `key`) to data key-value pairs. As a result, it can generate a YAML tree out of the code documentation and also a Markdown document.
```